catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/MetacardFactory.java

```diff
@@ -0,0 +1,128 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardCreationException;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.impl.FrameworkProperties;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.security.Subject;
+import ddf.security.SubjectUtils;
+
+/**
+ * Support class for creating metacards for the {@code CatalogFrameworkImpl}.
+ *
+ * This factory class contains methods specific to metacard creation for the CFI and its
+ * support classes. No operations/support methods should be added to this class except in support
+ * of CFI metacard creation.
+ */
+public class MetacardFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetacardFactory.class);
+
+    //
+    // Injected properties
+    //
+    private final FrameworkProperties frameworkProperties;
+
+    public MetacardFactory(FrameworkProperties frameworkProperties) {
+        this.frameworkProperties = frameworkProperties;
+    }
+
+    Metacard generateMetacard(String mimeTypeRaw, String id, String fileName, Subject subject,
+            Path tmpContentPath) throws MetacardCreationException, MimeTypeParseException {
+
+        Metacard generatedMetacard = null;
+        InputTransformer transformer = null;
+        StringBuilder causeMessage = new StringBuilder("Could not create metacard with mimeType ");
+        try {
+            MimeType mimeType = new MimeType(mimeTypeRaw);
+
+            List<InputTransformer> listOfCandidates =
+                    frameworkProperties.getMimeTypeToTransformerMapper()
+                            .findMatches(InputTransformer.class, mimeType);
+
+            LOGGER.debug("List of matches for mimeType [{}]: {}", mimeType, listOfCandidates);
+
+            for (InputTransformer candidate : listOfCandidates) {
+                transformer = candidate;
+
+                try (InputStream transformerStream = com.google.common.io.Files.asByteSource(
+                        tmpContentPath.toFile())
+                        .openStream()) {
+                    generatedMetacard = transformer.transform(transformerStream);
+                }
+                if (generatedMetacard != null) {
+                    break;
+                }
+            }
+        } catch (CatalogTransformerException | IOException e) {
+            causeMessage.append(mimeTypeRaw)
+                    .append(". Reason: ")
+                    .append(System.lineSeparator())
+                    .append(e.getMessage());
+
+            // The caught exception more than likely does not have the root cause message
+            // that is needed to inform the caller as to why things have failed.  Therefore
+            // we need to iterate through the chain of cause exceptions and gather up
+            // all of their message details.
+            Throwable cause = e.getCause();
```

Comment:
Feels like a good candidate for an `extractRootCause()` private method and kill the comment :-)